modules/seqexec/server/src/main/scala/seqexec/server/tcs/Gaos.scala

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server.tcs
+
+import seqexec.server.SeqActionF
+import seqexec.server.tcs.TcsController.FocalPlaneOffset
+import squants.Time
+
+/*
+ * Interface to control AO systems. Implemented by Altair and GeMS
+ */
+trait Gaos {
+  import Gaos._
+
+  /*
+   * Pause GAOS guiding. The GAOS system decides what to pause, according to the reasons given
+   */
+  def pause[F[_]](reasons: Set[Reason]): SeqActionF[F, Unit]
```

Comment:
Should we consider simplifying this. In principle it should be possible to directly return `F[Unit]` Of course the worry is that this may ripple across the codebase making it a very big change but it would be interesting to explore it

Reply:
`SeqActionF` is an `EitherT[F, SeqexecFailure, A]`, and we are very interested in the failures.

Reply:
Hmm I don't think it's possible to implement a method like

```scala
def pause[F[_]](reasons: Set[Reason]): SeqActionF[F, Unit]
```

Without constraints on `F`. I think maybe consider

```scala
trait Gaos[F[_]] {
  def pause(reasons: Set[Reason]): F[Unit]
  def resume(reasons: Set[Reason]): F[Unit]
  ...
}
```

And then you can demand `F[_]: MonadError[?[_], Throwable]: Sync` to construct a concrete `Gaos[F]`. This is called tagless style and it's probably a best practice these days.

Reply:
Why do you need `MonadError`?

Reply:
If the implementation wants to `raiseError(blah)` then you need at least `ApplicativeError[F, SeqError]` but maybe you don't need to do this. I was just guessing based on the original signature.

Reply:
Se me bay just use `F[_]` here an at the implementation level put the minimal constrains to make it work